diff --git a/dom/media/moz.build b/dom/media/moz.build
index 354bba2e996f..9ad9cfa24507 100644
--- a/dom/media/moz.build
+++ b/dom/media/moz.build
@@ -25,13 +25,15 @@ component_av = ("Core", "WebRTC: Audio/Video")
 with Files("GetUserMedia*"):
     BUG_COMPONENT = component_av
 
+if CONFIG["MOZ_WEBRTC"]:
+    DIRS += ["bridge"]
+
 DIRS += [
     "autoplay",
     "doctor",
     "driftcontrol",
     "eme",
     "encoder",
-    "fake-cdm",
     "flac",
     "gmp",
     "gmp-plugin-openh264",
@@ -60,9 +62,6 @@ DIRS += [
 if CONFIG["MOZ_ANDROID_HLS_SUPPORT"]:
     DIRS += ["hls"]
 
-if CONFIG["MOZ_WEBRTC"]:
-    DIRS += ["bridge"]
-
 if CONFIG["MOZ_WEBSPEECH"]:
     DIRS += ["webspeech"]
 
